### PR TITLE
Chat branch - Cookies, Logging In, On-boarding, Updating a User

### PR DESCRIPTION
**Pull request by innabobolina**
Set up cookies, added login in functionality (Mongo .findOne() method), on-boarding and updating a user functionalities (Mongo .updateOne() method).
Tested that everything is working properly and that Mongo database is being properly updated. Registering new users and logging in existing users and storing cookies works properly on both front end and back end.

Note: instead of running "npm start", run commands separately for client and server directories:
    client: "npm run start:frontend"
    server: "npm run start:backend"

image

### CODE REVIEW

File: client/src/components/AuthModal.js [hunk 1]
```diff
@@ -24,14 +26,20 @@ const AuthModal = ({setShowModal, isSignUp}) => {
                 setError('Passwords need to match!');
                 return;
             }
-            console.log('make a post request to our database');
-            const response = await axios.post("http://localhost:8000/signup", {email, password});
+            console.log('make a post request to our database', email, password);
```
Comment (Zyanvel) [inline client/src/components/AuthModal.js:29]: Hi Inna! I am assuming these console.log has a specific purpose for the backend or can it be removed? thank you

File: client/src/components/AuthModal.js [hunk 2]
```diff
@@ -24,14 +26,20 @@ const AuthModal = ({setShowModal, isSignUp}) => {
...
+            // const response = await axios.post("http://localhost:8000/signup", {email, password});
```
Comment (Zyanvel) [inline client/src/components/AuthModal.js:30]: Please remove unused code, thank you!

File: client/src/components/AuthModal.js [hunk 3]
```diff
@@ -24,14 +26,20 @@ const AuthModal = ({setShowModal, isSignUp}) => {
...
+            const response = await axios.post(`http://localhost:8000/${isSignUp ? 'signup' : 'login'}`, { email, password });
 
-            const success = response.status === 201;
-            if (success) navigate ('/onboarding');
+            // setCookie('Email', response.data.email);
```
Comment (Zyanvel) [inline client/src/components/AuthModal.js:33]: Please remove unused code (if applicable) thank you!

File: client/src/components/AuthModal.js [hunk 4]
```diff
@@ -24,14 +26,20 @@ const AuthModal = ({setShowModal, isSignUp}) => {
...
+            setCookie('UserId', response.data.userId);
+            setCookie('AuthToken', response.data.token);
+
+            console.log(response);
 
-            // if (success && isSignUp) navigate ('/onboarding')
```
Comment (Zyanvel) [inline client/src/components/AuthModal.js:33]: Please remove unused code (if applicable) thank you!

File: client/src/pages/Onboarding.js [hunk 5]
```diff
@@ -1,11 +1,16 @@
...
+import axios from 'axios';
 import Nav from '../components/Nav';
 import '../index.css';
 
 const Onboarding = () => {
+  const [ cookies, setCookie, removeCookie] = useCookies(['user']);
 
   const [formData, setFormData] = useState({
+    user_id: cookies.UserId,
+    // email: cookies.Email,
```
Comment (Zyanvel) [inline client/src/pages/Onboarding.js:13]: Is this a comment or maybe unused code? Thank you!

**Review by Zyanvel — COMMENTED** [review]
General looks good to me, I just left some comments regarding console.logs and what it seemed unused code commented out (or perhaps there were comments on the code).

**Comment by innabobolina** [comment]
Hi Zyan,Thanks so much for reviewing and commenting.
As for the console.logs, I prefer to keep them for now because they’re super helpful in development. They allow me to understand what is going on and to learn. :) I will remove them all for production when the site is ready. Similarly for some alternative code that is commented out for now, it helps understand the logic and steps.
Don’t worry, I will remove them all at a later point. :)

Sent from Yahoo Mail for iPhone


On Tuesday, August 30, 2022, 4:34 PM, Zyanvel ***@***.***> wrote:




@Zyanvel commented on this pull request.

General looks good to me, I just left some comments regarding console.logs and what it seemed unused code commented out (or perhaps there were comments on the code).

**Review by Zyanvel — APPROVED** [review]

**Comment by innabobolina** [comment]
Thanks so much! :)

    On Tuesday, August 30, 2022 at 05:52:35 PM EDT, Zyanvel ***@***.***> wrote:  
 
 


@Zyanvel approved this pull request.

**Review by ramblelob — APPROVED** [review]

**Pull request merged**